pkpdapp/pkpdapp/views/data_analysis.py

```diff
@@ -0,0 +1,83 @@
+#
+# This file is part of PKPDApp (https://github.com/pkpdapp-team/pkpdapp) which
+# is released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+from pkpdapp.models import (
+    Biomarker, BiomarkerType, Dose, Protocol
+)
+import pandas as pd
+from django.views.generic import (
+    TemplateView
+)
+from pkpdapp.dash_apps.data_analysis_app import DataAnalysisApp
+from django.contrib.auth.mixins import LoginRequiredMixin
+
+
+def create_data_analysis_app(project):
+    datasets = []
+    dataset_names = []
+    for dataset in project.datasets.all():
+        protocol = Protocol.objects.filter(dataset=dataset)
+        if not protocol:
+            continue
+        biomarker_types = BiomarkerType.objects.filter(
+            dataset=dataset,
+        )
+        biomarkers = Biomarker.objects\
+            .select_related('biomarker_type__name')\
+            .filter(biomarker_type__in=biomarker_types)
+
+        # convert to pandas dataframe with the column names expected
+        df_meas = pd.DataFrame(
+            list(
+                biomarkers.values('time', 'subject_id',
+                                  'biomarker_type__name', 'value')
+            )
+        )
+        doses = Dose.objects.filter(protocol__in=protocol)\
+                    .select_related('protocol__subject_id')\
+                    .select_related('protocol__compound__name')
+
+        df_dose = pd.DataFrame(
+            list(
+                doses.values('start_time', 'protocol__subject_id',
+                             'protocol__compound__name', 'amount')
+            )
+        )
+        df_dose.rename(columns={
+            'protocol__subject_id': 'subject_id',
+        }, inplace=True)
+
+        initial_doses = df_dose.sort_values(by='start_time')\
+                               .groupby(['subject_id'])\
+                               .first()\
+                               .drop(columns='start_time')
+
+        df = pd.merge(
+            df_meas, initial_doses,
+            how='inner', on=['subject_id']
+        )
+
+        df.rename(columns={
+            'subject_id': 'ID',
+            'time': 'Time',
+            'protocol__compound__name': 'Compound',
+            'value': 'Measurement',
+            'biomarker_type__name': 'Biomarker',
+            'amount': 'Amount'
+        }, inplace=True)
```

Comment:
I'm guessing this will go once we standardise names for biomarkers vs Roche's list of dataset columns?

Reply:
it will change, but you'll always need to change names like `protocol__compound__name`